src/simplify_qdq.cpp

```diff
@@ -0,0 +1,205 @@
+#include <migraphx/simplify_qdq.hpp>
+#include <migraphx/instruction.hpp>
+#include <migraphx/iterator_for.hpp>
+#include <migraphx/make_op.hpp>
+#include <migraphx/program.hpp>
+#include <migraphx/shape.hpp>
+#include <migraphx/matcher.hpp>
+#include <migraphx/dead_code_elimination.hpp>
+#include <migraphx/pass_manager.hpp>
+#include <migraphx/op/convolution.hpp>
+#include <migraphx/op/quant_convolution.hpp>
+#include <migraphx/op/dot.hpp>
+#include <migraphx/op/quant_dot.hpp>
+#include <migraphx/register_op.hpp>
+
+namespace migraphx {
+inline namespace MIGRAPHX_INLINE_NS {
+
+std::unordered_set<std::string> get_quantizable_op_names()
+{
+    static std::unordered_set<std::string> s = {"convolution", "dot"};
+    return s;
+}
+
+std::unordered_set<std::string> get_all_op_names()
+{
+    static auto ops = get_operators();
+    static std::unordered_set<std::string> s(ops.begin(), ops.end());
+    return s;
+}
+
+bool inputs_are_zeros(const instruction_ref& ins)
+{
+    literal zp_lit;
+    if(ins->name() == "multibroadcast" or ins->name() == "broadcast")
+        zp_lit = ins->inputs().at(0)->get_literal();
+    else
+        zp_lit = ins->get_literal();
+    std::vector<int64_t> zero_points;
+    zp_lit.visit([&](const auto zp) {
+        std::transform(
+            zp.begin(), zp.end(), std::back_inserter(zero_points), [](auto&& z) { return z; });
+    });
+    return std::all_of(
+        zero_points.begin(), zero_points.end(), [](const auto& z) { return z == 0; });
+}
+
+double get_scale(const instruction_ref& ins)
```

Comment:
This should be done with the matcher. You can use `skip_broadcasts` to skip the broadcasts and then bind the instruction.